IanGRap/Squish
# Fix 2 bugs in player select lobby

Contributors: Austin

Goal: Fix 2 bugs in the player select lobby
1) Sometimes, even if all players are ready, the begin match screen won't pop up (we saw this in the playtest today)
2) When players collide with each other in the lobby, a console error is thrown

What changed:
- In ActivePlayers, before adding a new player, ensure that it does not already exist in the list (prevents duplicates)
- In PlayerSelectUIManager, changed the way I check whether the Begin Rave screen should be shown. It is now more strict.
- In PlayerAdjacencyChecker, don't try to use info from the Spawner.Instance if it does not exist

How to Test:
- To check bugfix 1:
    - Go to the lobby, sign in and out many times, die, generally do a lot. Make sure that every time all of the players are ready, the Begin Rave screen appears
- To check bugfix 2:
    - Go to the lobby and collide with another player. Make sure there are no console errors.
